packages/core/strapi/lib/services/entity-validator/index.js

```diff
@@ -222,10 +223,121 @@ const createValidateEntity =
         entity,
       },
       { isDraft }
-    ).required();
+    )
+      .test('relations-test', 'check that all relations exist', async function (data) {
+        try {
+          await checkRelationsExist(buildRelationsStore({ uid: model.uid, data }));
+        } catch (e) {
+          return this.createError({
+            path: this.path,
+            message: e.message,
+          });
+        }
+        return true;
+      })
+      .required();
+
     return validateYupSchema(validator, { strict: false, abortEarly: false })(data);
   };
 
+/**
+ * Builds an object containing all the media and relations being associated with an entity
+ * @param {String} uid of the model
+ * @param {Object} data
+ * @param {Object} relationsStore to be updated and returned
+ * @returns
+ */
+const buildRelationsStore = ({ uid, data, relationsStore = {} }) => {
+  if (!uid || !data) {
+    return relationsStore;
+  }
```

Comment:
I think it should throw if the uid is not provided, not silently return. What do you think?
Maybe just replacing this `if` by `if (isEmpty(data)) { return relationsStore }`.
If the uid is not provided or incorrect it will throw later anyway.